### PR TITLE
DOC: fix PR02 errors in docstring for pandas.core.groupby.DataFrameGroupBy.corrwith

### PR DESCRIPTION
**Pull request by jordan-d-murphy**
All PR02 Errors resolved in the following cases:

1. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.core.groupby.DataFrameGroupBy.corrwith
2. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.DataFrame.corrwith               

OUTPUT:

1. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.core.groupby.DataFrameGroupBy.corrwith
```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.core.groupby.DataFrameGroupBy.corrwith" correct. :)
```

2. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.DataFrame.corrwith    
```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.corrwith" correct. :)
```           

- [x] xref https://github.com/pandas-dev/pandas/issues/57111
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -2491,14 +2491,79 @@ def dtypes(self) -> Series:
...
+
+        >>> y = df2.groupby("A").first()
+        >>> y
+            B
+        A
+        1  2.4
+        2  2.1
+        3  2.5
+
+        >>> x.corrwith(y)
```
Comment (rhshadrach) [inline pandas/core/groupby/generic.py:2563]: Can you show example of using `groupby(...).corrwith`? `x` here is a Series, and so this is using `Series.corrwith`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR. Just in case you aren't aware, the validation runs on the CI, so you don't need to show the result of `OUTPUT:` if you don't want to - we'll see it if it fails :smile:.

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jordan-d-murphy) [reply pandas/core/groupby/generic.py:2563]: You bet! Thanks for catching that; I'll update the example and push a new commit.

Comment (jordan-d-murphy) [reply pandas/core/groupby/generic.py]: Pushed a new code example, @rhshadrach can you take another look please?

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by jordan-d-murphy** [comment]
@mroeschke can you take a look at this PR when you get a chance?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks keep em coming!

**Pull request merged**